File: common/replace/replace.py
from argparse import ArgumentParser
import os
import re
import shutil

global opts
opts = {}

def parseOptions():
    argparser = ArgumentParser()
    argparser.add_argument('--file', help=':specify replace file')
    argparser.add_argument('--pre',  help=':specify pre regular expression')
    argparser.add_argument('--post', help=':specify post regular expression')
    args = argparser.parse_args()
    if args.file: opts.update({'file':args.file})
    if args.pre: opts.update({'pre':args.pre})
    if args.post: opts.update({'post':args.post})

def replace():
    file_path = os.path.abspath(opts['file'])
    temp_file = file_path + '.tmp'
    match = 0
    ofs = open(temp_file, mode='w', encoding='utf-8')
    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
        pattern = '.*' + opts['pre'] + '.*'
        for line in file:
            if re.match(pattern, line):
                match = 1
            line = re.sub(opts['pre'], opts['post'], line)
            ofs.write(line)
    ofs.close()
    if match:
        shutil.copymode(file_path, temp_file)
        shutil.move(temp_file, file_path)
    else:
        os.remove(temp_file)
    
if __name__ == '__main__':
    parseOptions()
    if (opts.get('file') and opts.get('pre') and opts.get('post')):
        replace()

File: common/replace/replace.pl
#!/usr/bin/perl -w

use strict;
use warnings;
use Getopt::Long qw(:config posix_default no_ignore_case bundling);
use Cwd;
use File::Find;
use File::Copy;
use Fcntl ':mode';

my %opts = ();
&parseOptions();

my $m_fmatch = '';
my $m_pre = '';
my $m_post = '';

sub parseOptions {
    GetOptions(\%opts, 'help',
                       'replace=s',
                       'file=s',
                       'pre=s',
                       'post=s',
    );
    if ($opts{help}) { &help(); exit; }
    &main();
}

sub help {
    print << "ENDLINE";
Usage  : ${0} [Options]
Option :
    --help   : print this usage
    --replace: specify replace execute path
    --file   : specify replace file regular expression 
    --pre    : specify pre regular expression 
    --post   : specify post regular expression 
ENDLINE
}

sub main() {
    if ($opts{replace}) {
        $m_fmatch = $opts{file}; $m_pre = $opts{pre}; $m_post = $opts{post};
        File::Find::find(\&replace, Cwd::realpath($opts{replace}));
    }
}

sub replace() {
    my $file_path = $File::Find::name;
    my $temp_file = "$file_path.tmp";
    my $match = 0;
    if ($file_path =~ /$m_fmatch/) {
        open (INFILE, "<$file_path") or die "$file_path: $!";
        open (OUTFILE, ">$temp_file") or die "$temp_file: $!";
        while (<INFILE>) {
            if (/$m_pre/) { $match = 1; }
            s/$m_pre/$m_post/g;
            print OUTFILE $_;
        }
        close (OUTFILE);
        close (INFILE);
        if ($match) {
            overwriteFile($file_path, $temp_file);
        }
        else {
            unlink($temp_file);
        }
    }
}

sub overwriteFile() {
    my $file_path = shift;
    my $temp_file = shift;

    my $mode = (stat($file_path))[2] & 07777;
    chmod $mode, $temp_file or die "Cannot chmod $temp_file: $!";

    File::Copy::move($temp_file, $file_path) or die "Cannot move $temp_file to $file_path: $!";
}
